server/src/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -476,6 +476,25 @@ public VolumeVO doInTransaction(TransactionStatus status) {
         });
     }
 
+    /**
+     * Retrieves the volume name from CreateVolumeCmd object.
+     *
+     * If the retrieved volume name is null, empty or blank, then A random name
+     * will be generated using getRandomVolumeName method.
+     *
+     * @param cmd
+     * @return Either the retrieved name or a random name.
+     */
+    public String getVolumeNameFromCommand(CreateVolumeCmd cmd) {
+        String userSpecifiedName = cmd.getVolumeName();
+
+        if (org.apache.commons.lang.StringUtils.isBlank(userSpecifiedName)) {
```

Comment:
Cool...then it all looks good to me.

Unfortunately, at the time being, I have my dev system in a state that would make it difficult and time consuming for me to test this.

If you do require a tester, please let me know. It might take a few days, though, before I finish up my current project and can put this code on my system.

Thanks